======================================================================
ERROR: Iteratively Matches 2 Length Array
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 248, in create_program
    weights.append(10 - population[i].score)
src.error.CaseyTimeoutError: Process exceeded 75 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   31.34
              create_program |         1 |    7.54
           random._randbelow |   1619697 |    4.72
                 list.append |   8370943 |    4.58
                   crossover |    321900 |    4.30
                         len |   7154710 |    4.11
            random.randrange |   1619697 |    3.83
              random.randint |   1619697 |    2.52
                      mutate |    643800 |    2.12
  _random.Random.getrandbits |   2850482 |    2.04
                         sum |    644800 |    1.97
               score_fitness |    644800 |    1.27
              int.bit_length |   1619697 |    1.01
           random.<listcomp> |      6438 |    0.89
              bisect._bisect |    643800 |    0.83
                         abs |   1289600 |    0.75
                    __init__ |    644800 |    0.53
       _random.Random.random |    643800 |    0.37
              random.choices |      6438 |    0.27
     generate_random_program |      1000 |    0.02
                       print |      9197 |    0.01
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 259, in create_program
    new_programs = crossover(n_p)
  File "biogimmickry.py", line 151, in crossover
    program1_seq = mutate(program1_seq)
src.error.CaseyTimeoutError: Process exceeded 75 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   34.39
                         len |  12878487 |    7.35
              create_program |         1 |    6.29
           random._randbelow |   1301073 |    4.05
                 list.append |   6715418 |    3.78
                   crossover |    258237 |    3.63
            random.randrange |   1301073 |    3.19
              random.randint |   1301073 |    2.07
  _random.Random.getrandbits |   2512774 |    1.88
                      mutate |    516472 |    1.77
                         sum |    517472 |    1.64
               score_fitness |    517472 |    1.08
              int.bit_length |   1301073 |    0.84
           random.<listcomp> |      5165 |    0.72
              bisect._bisect |    516500 |    0.69
                         abs |   1034944 |    0.63
                    __init__ |    517472 |    0.44
       _random.Random.random |    516500 |    0.30
              random.choices |      5165 |    0.22
     generate_random_program |      1000 |    0.02
                       print |      8892 |    0.01
----------------------------------------------------------------------


======================================================================
ERROR: Matches 1 Length Array
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 259, in create_program
    new_programs = crossover(n_p)
  File "biogimmickry.py", line 151, in crossover
    program1_seq = mutate(program1_seq)
src.error.CaseyTimeoutError: Process exceeded 15 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |    6.99
                         len |   3045520 |    1.69
              create_program |         1 |    1.24
           random._randbelow |    263922 |    0.78
                 list.append |   1315224 |    0.72
                   crossover |     50538 |    0.69
            random.randrange |    263922 |    0.63
              random.randint |    263922 |    0.42
  _random.Random.getrandbits |    478993 |    0.34
                      mutate |    101074 |    0.33
                         sum |    102074 |    0.25
               score_fitness |    102074 |    0.21
              int.bit_length |    263922 |    0.16
           random.<listcomp> |      1011 |    0.14
              bisect._bisect |    101100 |    0.13
                    __init__ |    102074 |    0.09
       _random.Random.random |    101100 |    0.06
              random.choices |      1011 |    0.06
                         abs |    102074 |    0.06
     generate_random_program |      1000 |    0.02
                       print |     10236 |    0.01
----------------------------------------------------------------------


======================================================================
ERROR: Matches 2-4 Length Non-Negative Array
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 270, in create_program
    new_population.append(new_programs[0])
src.error.CaseyTimeoutError: Process exceeded 30 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   13.96
                         len |   6190874 |    3.39
              create_program |         1 |    2.40
           random._randbelow |    525338 |    1.55
                 list.append |   2675068 |    1.42
                   crossover |    102845 |    1.35
            random.randrange |    525338 |    1.23
              random.randint |    525338 |    0.81
  _random.Random.getrandbits |    980752 |    0.69
                      mutate |    205690 |    0.66
                         sum |    206690 |    0.61
               score_fitness |    206690 |    0.41
              int.bit_length |    525338 |    0.32
           random.<listcomp> |      2057 |    0.27
              bisect._bisect |    205700 |    0.26
                         abs |    413380 |    0.24
                    __init__ |    206690 |    0.17
              random.choices |      2057 |    0.12
       _random.Random.random |    205700 |    0.11
     generate_random_program |      1000 |    0.02
                       print |     10113 |    0.01
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 259, in create_program
    new_programs = crossover(n_p)
  File "biogimmickry.py", line 152, in crossover
    program2_seq = mutate(program2_seq)
src.error.CaseyTimeoutError: Process exceeded 30 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   13.87
                         len |   5929097 |    3.25
              create_program |         1 |    2.31
           random._randbelow |    520124 |    1.54
                 list.append |   2647562 |    1.41
                   crossover |    101761 |    1.33
            random.randrange |    520124 |    1.21
                         sum |    204520 |    0.85
              random.randint |    520124 |    0.80
  _random.Random.getrandbits |   1004115 |    0.70
                      mutate |    203522 |    0.65
                         abs |    818080 |    0.45
               score_fitness |    204520 |    0.40
              int.bit_length |    520124 |    0.32
           random.<listcomp> |      2036 |    0.27
              bisect._bisect |    203600 |    0.25
                    __init__ |    204520 |    0.17
       _random.Random.random |    203600 |    0.11
              random.choices |      2036 |    0.10
     generate_random_program |      1000 |    0.02
                       print |     10319 |    0.01
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 262, in create_program
    score2 = new_programs[1].score_fitness(fe)
  File "biogimmickry.py", line 118, in score_fitness
    evaluated_score = fe.evaluate(self.sequence)
  File "biogimmickry.py", line 31, in <lambda>
    lambda program: FitnessEvaluator._evaluate(array, program)
  File "biogimmickry.py", line 100, in _evaluate
    actual = FitnessEvaluator.interpret(program, len(expect))
  File "biogimmickry.py", line 47, in interpret
    while p_ptr < len(program):
src.error.CaseyTimeoutError: Process exceeded 30 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   14.27
                         len |   6724912 |    3.64
              create_program |         1 |    2.28
           random._randbelow |    503581 |    1.41
                 list.append |   2563010 |    1.36
                   crossover |     98502 |    1.28
            random.randrange |    503581 |    1.16
              random.randint |    503581 |    0.77
                         sum |    198003 |    0.70
                      mutate |    197004 |    0.62
  _random.Random.getrandbits |    867853 |    0.61
               score_fitness |    198004 |    0.38
                         abs |    594009 |    0.33
              int.bit_length |    503581 |    0.30
           random.<listcomp> |      1971 |    0.26
              bisect._bisect |    197100 |    0.24
                    __init__ |    198004 |    0.16
       _random.Random.random |    197100 |    0.11
              random.choices |      1971 |    0.11
     generate_random_program |      1000 |    0.02
                       print |     10071 |    0.01
----------------------------------------------------------------------


======================================================================
ERROR: Matches 5-8 Length Array
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 259, in create_program
    new_programs = crossover(n_p)
UnboundLocalError: local variable 'sb' referenced before assignment
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 248, in create_program
    weights.append(10 - population[i].score)
src.error.CaseyTimeoutError: Process exceeded 45 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   18.77
              create_program |         1 |    4.22
           random._randbelow |    919205 |    2.78
                 list.append |   4725487 |    2.56
                   crossover |    181700 |    2.37
            random.randrange |    919205 |    2.16
                         len |   3676765 |    2.11
                         sum |    364400 |    1.98
              random.randint |    919205 |    1.41
  _random.Random.getrandbits |   1836777 |    1.30
                         abs |   2186400 |    1.21
                      mutate |    363400 |    1.18
               score_fitness |    364400 |    0.72
              int.bit_length |    919205 |    0.57
           random.<listcomp> |      3634 |    0.50
              bisect._bisect |    363400 |    0.46
                    __init__ |    364400 |    0.30
       _random.Random.random |    363400 |    0.21
              random.choices |      3634 |    0.16
     generate_random_program |      1000 |    0.02
                       print |      9705 |    0.01
----------------------------------------------------------------------
Traceback (most recent call last):
  File "biogimmickry.py", line 262, in create_program
    score2 = new_programs[1].score_fitness(fe)
  File "biogimmickry.py", line 118, in score_fitness
    evaluated_score = fe.evaluate(self.sequence)
  File "biogimmickry.py", line 31, in <lambda>
    lambda program: FitnessEvaluator._evaluate(array, program)
  File "biogimmickry.py", line 100, in _evaluate
    actual = FitnessEvaluator.interpret(program, len(expect))
src.error.CaseyTimeoutError: Process exceeded 45 second(s)

                    Function |     Calls | Seconds
  ------------------------------------------------
            [casey overhead] |         0 |   21.31
                         len |   9374917 |    5.17
              create_program |         1 |    3.36
           random._randbelow |    719222 |    2.11
                 list.append |   3683796 |    1.98
                   crossover |    141633 |    1.91
            random.randrange |    719222 |    1.69
                         sum |    284265 |    1.36
              random.randint |    719222 |    1.11
                      mutate |    283266 |    0.92
  _random.Random.getrandbits |   1267627 |    0.92
                         abs |   1421325 |    0.80
               score_fitness |    284266 |    0.57
              int.bit_length |    719222 |    0.45
           random.<listcomp> |      2833 |    0.38
              bisect._bisect |    283300 |    0.36
                    __init__ |    284266 |    0.24
       _random.Random.random |    283300 |    0.16
              random.choices |      2833 |    0.14
     generate_random_program |      1000 |    0.02
                       print |     10057 |    0.01
----------------------------------------------------------------------

                Matches 1 Length Array |  50% [15.0]
 Matches 2-4 Length Non-Negative Array |   0% [20.0]
              Matches 5-8 Length Array |  25% [30.0]
    Iteratively Matches 2 Length Array |   0% [35.0]
----------------------------------------------------
                               [TOTAL] |  15%

------------------------------
 Submission Status | SUCCESS |
------------------------------
